src/EA.Weee.RequestHandlers/AatfReturn/ObligatedReused/GetAatfSiteDataAccess.cs

```diff
@@ -0,0 +1,38 @@
+﻿namespace EA.Weee.RequestHandlers.AatfReturn.ObligatedReused
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Data.Entity;
+    using System.Linq;
+    using System.Threading.Tasks;
+    using EA.Weee.DataAccess;
+    using EA.Weee.Domain.AatfReturn;
+    using EA.Weee.RequestHandlers.AatfReturn.ObligatedGeneric;
+    using EA.Weee.RequestHandlers.AatfReturn.Specification;
+
+    internal class GetAatfSiteDataAccess : IGetAatfSiteDataAccess
+    {
+        private readonly WeeeContext context;
+        private readonly IGenericDataAccess genericDataAccess;
+
+        public GetAatfSiteDataAccess(WeeeContext context, IGenericDataAccess genericDataAccess)
+        {
+            this.context = context;
+            this.genericDataAccess = genericDataAccess;
+        }
+
+        public async Task<List<AatfAddress>> GetAddresses(Guid aatfId, Guid returnId)
+        {
+            var weeeReusedId = (await genericDataAccess.GetManyByExpression<WeeeReused>(new WeeeReusedByAatfIdAndReturnIdSpecification(aatfId, returnId))).Last().Id;
```

Comment:
We can also write unit tests for this without checking the database... check ReturnSchemeDataAccessTests for examples